File: selenium-test-suite/tests/sdk-audit-init-and-basic-reports.test.js
/**
 * AccessFlow JS SDK — Behavior Tests (Selenium)
 *
 * Validates SDK initialization, audit lifecycle, and report shape
 * using Selenium WebDriver. Mirrors the Playwright sdk-behavior.spec.js.
 */

const { AccessFlowSDK, SeleniumDriver } = require('@acsbe/accessflow-sdk');
const { createDriver, navigateTo } = require('../helpers/driver');

AccessFlowSDK.init({
  apiKey:
    process.env.AF_NODE_PACKAGE_KEY ||
    process.env.AF_Node_Package_Key ||
    'flow-1TmAHDBGjDGX4aadlA0000VdvvxVnhZfP2',
});

let driver;

beforeAll(async () => {
  driver = await createDriver();
});

afterAll(async () => {
  if (driver) await driver.quit();
});

describe('SDK Initialization', () => {
  test('should initialize without errors', () => {
    expect(true).toBe(true);
  });

  test('should create an instance bound to a driver', async () => {
    await navigateTo(driver, '/');
    const sdk = new AccessFlowSDK(new SeleniumDriver(driver));
    expect(sdk).toBeTruthy();
  });
});

describe('Audit — Home Page', () => {
  beforeEach(async () => {
    await navigateTo(driver, '/');
    await driver.sleep(500);
  });

  test('should return a report on the home page', async () => {
    const sdk = new AccessFlowSDK(new SeleniumDriver(driver));
    const report = await sdk.audit();
    expect(report).toBeTruthy();
  });

  test('audit should complete without throwing', async () => {
    const sdk = new AccessFlowSDK(new SeleniumDriver(driver));
    await expect(sdk.audit()).resolves.toBeTruthy();
  });
});

describe('Audit — Graphics Route', () => {
  beforeEach(async () => {
    await navigateTo(driver, '/graphics/alt-text_success');
    await driver.sleep(500);
  });

  test('should return a report for the graphics/alt-text route', async () => {
    const sdk = new AccessFlowSDK(new SeleniumDriver(driver));
    const report = await sdk.audit();
    expect(report).toBeTruthy();
  });

  test('multiple audits on the same page should be stable', async () => {
    const sdk = new AccessFlowSDK(new SeleniumDriver(driver));
    const reportA = await sdk.audit();
    const reportB = await sdk.audit();
    expect(reportA).toBeTruthy();
    expect(reportB).toBeTruthy();
  });
});

describe('Audit — After Navigation', () => {
  test('should audit successfully after route change', async () => {
    await navigateTo(driver, '/');
    await driver.sleep(500);
    const sdkHome = new AccessFlowSDK(new SeleniumDriver(driver));
    const reportHome = await sdkHome.audit();
    expect(reportHome).toBeTruthy();

    await navigateTo(driver, '/graphics/alt-text_success');
    await driver.sleep(500);
    const sdkGraphics = new AccessFlowSDK(new SeleniumDriver(driver));
    const reportGraphics = await sdkGraphics.audit();
    expect(reportGraphics).toBeTruthy();
  });
});

File: sdk/tests/selenium/java/src/test/java/com/accessflow/SdkAuditInitAndBasicReportsTest.java
package com.accessflow;

import com.acsbe.accessflow.AccessFlowSDK;
import com.acsbe.accessflow.AccessFlowTeardown;
import com.acsbe.accessflow.SeleniumDriver;
import org.junit.jupiter.api.*;
import org.openqa.selenium.WebDriver;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

/**
 * AccessFlow Java SDK — Init & Basic Reports (Selenium).
 * Mirrors Node sdk-audit-init-and-basic-reports.test.js.
 */
@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
public class SdkAuditInitAndBasicReportsTest {

    private WebDriver driver;

    @BeforeEach
    void setup() {
        driver = DriverHelper.createDriver();
    }

    @AfterEach
    void teardown() {
        if (driver != null) driver.quit();
    }

    @AfterAll
    static void finalizeReports() {
        LocalReportCollector.write();
        AccessFlowTeardown.finalizeReports();
    }

    private AccessFlowSDK createSdk() {
        String apiKey = DriverHelper.getApiKey();
        return apiKey != null && !apiKey.isEmpty()
                ? new AccessFlowSDK(new SeleniumDriver(driver), apiKey)
                : new AccessFlowSDK(new SeleniumDriver(driver));
    }

    @Test @Order(1) @DisplayName("SDK initializes without errors")
    void testInitWithoutErrors() {
        assertNotNull(driver);
    }

    @Test @Order(2) @DisplayName("SDK creates an instance bound to the Selenium driver")
    void testInstanceBindsToDriver() {
        DriverHelper.navigateTo(driver, "/");
        AccessFlowSDK sdk = createSdk();
        assertNotNull(sdk);
    }

    @Test @Order(3) @DisplayName("Audit returns a report on the home page")
    void testAuditHomePage() {
        DriverHelper.navigateTo(driver, "/");
        AccessFlowSDK sdk = createSdk();
        Map<String, Object> report = sdk.audit();
        AccessFlowTeardown.recordAudit(driver.getCurrentUrl(), report);
        LocalReportCollector.add(driver.getCurrentUrl(), report);
        assertNotNull(report);
    }

    @Test @Order(4) @DisplayName("Audit completes without throwing on the home page")
    void testAuditDoesNotThrow() {
        DriverHelper.navigateTo(driver, "/");
        AccessFlowSDK sdk = createSdk();
        assertDoesNotThrow(() -> {
            Map<String, Object> report = sdk.audit();
            AccessFlowTeardown.recordAudit(driver.getCurrentUrl(), report);
            LocalReportCollector.add(driver.getCurrentUrl(), report);
        });
    }

    @Test @Order(5) @DisplayName("Audit returns a report on the graphics/alt-text route")
    void testAuditGraphicsRoute() {
        DriverHelper.navigateTo(driver, "/graphics/alt-text_success");
        AccessFlowSDK sdk = createSdk();
        Map<String, Object> report = sdk.audit();
        AccessFlowTeardown.recordAudit(driver.getCurrentUrl(), report);
        LocalReportCollector.add(driver.getCurrentUrl(), report);
        assertNotNull(report);
    }

    @Test @Order(6) @DisplayName("Multiple audits on the same page are stable")
    void testMultipleAuditsStable() {
        DriverHelper.navigateTo(driver, "/graphics/alt-text_success");
        AccessFlowSDK sdk = createSdk();
        Map<String, Object> reportA = sdk.audit();
        Map<String, Object> reportB = sdk.audit();
        AccessFlowTeardown.recordAudit(driver.getCurrentUrl(), reportA);
        AccessFlowTeardown.recordAudit(driver.getCurrentUrl(), reportB);
        LocalReportCollector.add(driver.getCurrentUrl(), reportA);
        LocalReportCollector.add(driver.getCurrentUrl(), reportB);
        assertNotNull(reportA);
        assertNotNull(reportB);
    }

    @Test @Order(7) @DisplayName("Audit works correctly after in-app route change")
    void testAuditAfterRouteChange() {
        DriverHelper.navigateTo(driver, "/");
        AccessFlowSDK sdk = createSdk();
        Map<String, Object> reportHome = sdk.audit();
        AccessFlowTeardown.recordAudit(driver.getCurrentUrl(), reportHome);
        LocalReportCollector.add(driver.getCurrentUrl(), reportHome);
        assertNotNull(reportHome);

        DriverHelper.navigateTo(driver, "/graphics/alt-text_success");
        AccessFlowSDK sdk2 = createSdk();
        Map<String, Object> reportGraphics = sdk2.audit();
        AccessFlowTeardown.recordAudit(driver.getCurrentUrl(), reportGraphics);
        LocalReportCollector.add(driver.getCurrentUrl(), reportGraphics);
        assertNotNull(reportGraphics);
    }
}
